src/main/java/com/commercetools/paymenttoorderprocessor/jobs/actions/OrderCreator.java

```diff
@@ -30,41 +41,67 @@
  *
  */
-public class MessageWriter implements ItemWriter<Cart> {
-    public static final Logger LOG = LoggerFactory.getLogger(MessageWriter.class);
+public class OrderCreator implements ItemWriter<CartAndMessage> {
+    private static final Logger LOG = LoggerFactory.getLogger(OrderCreator.class);
 
     private static final String ENCRYPTIONALGORITHM = "Blowfish";
+    private static final int DEFAULTTIMEOUT = 40000;
 
     @Value("${createorder.encryptionkey}")
     private String encryptionKey;
     @Value("${createorder.endpoint.url}")
     private String urlstring;
+
+    @Autowired
+    private MessageProcessedManager messageProcessedManager;
+
+    @Autowired
+    HttpClient httpClient;
     
+    @Autowired
+    TimeStampManager timeStampManager;
+
     @Override
-    public void write(List<? extends Cart> items) throws Exception {
-        for (Cart item : items) {
+    public void write(List<? extends CartAndMessage> items) {
+        for (CartAndMessage item : items) {
             sendRequestToCreateOrder(item);
         }
     }
 
-    private void sendRequestToCreateOrder(Cart cart) throws Exception {
+    private void sendRequestToCreateOrder(CartAndMessage cartAndMessage) {
+        final Cart cart = cartAndMessage.getCart();
         final String body = SphereJsonUtils.toJsonString(cart);
+        //encrypting cart
         final String bodyEncrypt = encrypt(body);
+        if (bodyEncrypt == null) {
+            timeStampManager.processingMessageFailed();
+            return;
+        }
         final List<NameValuePair> headerList = new ArrayList<NameValuePair>();
         headerList.add(NameValuePair.of(HttpHeaders.CONTENT_TYPE,"text/plain"));
         headerList.add(NameValuePair.of("Content-Length", String.valueOf(bodyEncrypt.length())));
         final HttpHeaders httpHeader = HttpHeaders.of(headerList);
         final HttpRequestBody httpBody = StringHttpRequestBody.of(bodyEncrypt);
         final HttpRequest httpRequest = HttpRequest.of(HttpMethod.POST, urlstring, httpHeader, httpBody);
-        final HttpResponse httpResponse = httpClient.execute(httpRequest).toCompletableFuture().get(20000, TimeUnit.MILLISECONDS);
-        final Integer statusCode = httpResponse.getStatusCode();
-        if (statusCode != 200) {
-            LOG.warn("Got Http-StatusCode {} from CreateOrder Endpoint for Cart {}", statusCode, cart);
+        HttpResponse httpResponse;
+        try {
+            //sending encrypted cart to API
+            httpResponse = httpClient.execute(httpRequest).toCompletableFuture().get(DEFAULTTIMEOUT, TimeUnit.MILLISECONDS);
+            if (httpResponse.hasSuccessResponseCode()) {
+                messageProcessedManager.setMessageIsProcessed(cartAndMessage.getMessage());
+            }
+            else {
+                LOG.warn("Response Code from API was {}", httpResponse.getStatusCode());
+                messageProcessedManager.setMessageIsProcessed(cartAndMessage.getMessage());
```

Comment:
i've tested on staging:
if order creation API responded any error (bad request, internal server error etc):
 
1. an order is not created
2. message marked as _processed_
3. next calls will not fetch this method any more